### ISSUE TITLE
Current implementation failes to load CA certificates from npm config

### ISSUE THREAD

**Issue by vsenko** [Medium/phantomjs#444]
Commit 2620fb283439181cab0a99b89fdf4532455bc35d introduced a new way to obtain CA certificates (as well as other configuration parameters) from npm config storage.

New algorithm fails if **cafile** configuration option points to a file that contains several concatenated certificates. **cafile** is read as a Buffer and passed to **ca** option:
https://github.com/Medium/phantomjs/commit/2620fb283439181cab0a99b89fdf4532455bc35d#diff-f16acefe4b6553580c43edab685f50f3R188

It will only work if **cafile** contains only one certificate. As https://nodejs.org/api/tls.html#tls_tls_connect_port_host_options_callback states:

>ca: A string, Buffer or array of strings or Buffers of trusted certificates in PEM format. If this is omitted several well known "root" CAs will be used, like VeriSign. These are used to authorize connections.

So if **cafile** contains several certificates its content should not be passed as a single Buffer.

I suggest to read this file as a string and to slice it using regexp:
```javascript
  ca = fs.readFileSync(process.env.npm_config_cafile, {encoding: 'utf8'})
    .split(/\n(?=-----BEGIN CERTIFICATE-----)/g);
```

Tested on nodejs v4.2.6, npm 2.14.12 on Ubuntu 14.04.

P.S.:
The old approach took into account only **ca** npm configuration option (it seems that it got automatically populated with correct values from **cafile** if **cafile** was set):
https://github.com/Medium/phantomjs/commit/2620fb283439181cab0a99b89fdf4532455bc35d#diff-f16acefe4b6553580c43edab685f50f3L187
Currently **ca** option does not get populated from **cafile** automatically.

**Comment by nicks** [comment]
hmmm...yeah...this is what npmconf does
https://github.com/npm/npmconf/blob/d5167b419b7aa8137512a3f1cb8ba89c82906f5c/lib/load-cafile.js#L9

i think we would accept a pull request for this (we have no easy way to test it because we don't use cafile in this way)

**Comment by nicks** [comment]
We changed the name of the package to phantomjs-prebuilt, and it should be fixed in v2.1.3

**Comment by vsenko** [comment]
:+1:
